src/kits/kitsController.ts

```diff
@@ -276,6 +280,32 @@ export class KitsController {
             if (chosen_kit.kit.name === SpecialKits.ScanForKits) {
                 await KitsController.scanForKits(await this.project.getCMakePathofProject());
                 return false;
+            } else if (chosen_kit.kit.name === SpecialKits.ScanSpecificDir) {
+                const dir = await vscode.window.showOpenDialog({
+                    canSelectFiles: false,
+                    canSelectFolders: true,
+                    canSelectMany: false,
+                    openLabel: localize('select.folder', 'Select Folder')
+                });
+                if (!dir || dir.length === 0) {
+                    return false;
+                }
+                const dirPathWithDepth = async (folder: string, depth: number = 5) => {
```

Comment:
Confirming my understanding, this is a recursive method adding ALL directories under the selected folder to the paths to search?

Reply:
Nah, It has a depth limit of 5.